src/game_renderer/GameRenderer.cpp

```diff
@@ -185,15 +185,244 @@ void GameRenderer::RenderLevelEditor(GLFWwindow* window,
   RenderIt(window, game_state, true);
 }
 
-void GameRenderer::RenderIt(GLFWwindow* window,
+void GameRenderer::RenderMiniMap(GLFWwindow* window,
+                std::shared_ptr<GameState> game_state,
+                bool render_level_editor) {
```

Comment:
A way I could see consolidating it is pulling each type of item draw routine out into its own method. But that would be a lot of tiny functions.